code/modules/almayer/weaponhits.dm

```diff
@@ -69,21 +88,91 @@
 				for(var/turf/picked_atom in location)
 					if(prob(hitchance))
 						cell_explosion(picked_atom, 600, 600, EXPLOSION_FALLOFF_SHAPE_EXPONENTIAL, null, antishiprailgun_cause_data)
+						GLOB.ship_health_vars["ship_hull_health"] = GLOB.ship_health_vars["ship_hull_health"] - WEAPON_RAILGUN_DAMAGE
 						shakeship(5, 5, FALSE, FALSE)
 						confirmedhit += 1
+						logtotalhits += 1
 				if(confirmedhit > 0)
 					weaponhits_effects(WEAPON_RAILGUN)
 				if(confirmedhit < 1)
 					weaponhits_effects(WEAPON_RAILGUN, TRUE)
```

Comment:
```suggestion
						confirmedhit++
						logtotalhits++
				weaponhits_effects(WEAPON_RAILGUN, confirmedhit < 1)
```
If you don't intend to ever have multiple types of weapons hitting in one call, then you don't even need confirmedhit and instead could use logtotalhits.